aws/ecs/client.go

```diff
@@ -0,0 +1,68 @@
+package ecs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	awsConfig "github.com/aws/aws-sdk-go-v2/config"
+	awsECS "github.com/aws/aws-sdk-go-v2/service/ecs"
+)
+
+type Client struct {
+	awsECS.Client
+
+	ctx context.Context
+}
+
+func NewECSClient(ctx context.Context, region string) (*Client, error) {
+	cfg, err := awsConfig.LoadDefaultConfig(ctx,
+		awsConfig.WithRegion(region),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{Client: *awsECS.NewFromConfig(cfg), ctx: ctx}, nil
+}
+
+func (c *Client) GetTaskPrivateIP(clusterArn, serviceName string) (string, error) {
+	taskArn, err := c.getTaskArn(clusterArn, serviceName)
+	if err != nil {
+		return "", err
+	}
+	return c.getTaskPrivateIP(clusterArn, taskArn)
+}
+
+func (c *Client) getTaskArn(clusterArn, serviceName string) (string, error) {
+	taskList, err := c.ListTasks(c.ctx, &awsECS.ListTasksInput{
+		Cluster:     &clusterArn,
+		ServiceName: &serviceName,
+	})
+	if err != nil {
+		return "", err
+	}
+	if len(taskList.TaskArns) < 1 {
+		return "", errors.New("no task arn found")
+	}
+	return taskList.TaskArns[0], nil
```

Comment:
Can `c.ListTasks` returns more than one value?

Reply:
It should not as we deploy one single task to one single service, but the API response returns a list of tasks. I can return an error if there is more than one task as it would be unexpected.